lib/iris/aux_factory.py

```diff
@@ -296,7 +296,6 @@ def _remap_with_bounds(self, dependency_dims, derived_dims):
                     # extra dimension to make the shape compatible, so
                     # we just add an extra 1.
                     shape.append(1)
-                nd_values = np.array(nd_values)
```

Comment:
@corinnebosley I'm convinced that at this point, `nd_values` will always be an array (`dask` or `numpy`), as returned by `_nd_points` or `_nd_bounds`, and we've now added tests to implicitly check that, so I believe we're safe making this change (which is one of the sources of lazy data being realised)